geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/GCStatsMonitor.java

```diff
@@ -14,73 +14,76 @@
  */
 package org.apache.geode.management.internal.beans.stats;
 
+import java.util.concurrent.atomic.AtomicLong;
+
 import org.apache.geode.StatisticDescriptor;
 import org.apache.geode.internal.statistics.StatisticId;
 import org.apache.geode.internal.statistics.StatisticNotFoundException;
 import org.apache.geode.internal.statistics.StatisticsNotification;
 
 public class GCStatsMonitor extends MBeanStatsMonitor {
+  private AtomicLong collections = new AtomicLong(0);
+  private AtomicLong collectionTime = new AtomicLong(0);
 
-  private volatile long collections = 0;
-
-  private volatile long collectionTime = 0;
+  long getCollections() {
+    return collections.get();
+  }
 
+  long getCollectionTime() {
+    return collectionTime.get();
+  }
 
   public GCStatsMonitor(String name) {
     super(name);
   }
 
+  void decreasePrevValues(DefaultHashMap statsMap) {
+    collections.set(collections.get() - statsMap.get(StatsKey.VM_GC_STATS_COLLECTIONS).intValue());
+    collectionTime
+        .set(collectionTime.get() - statsMap.get(StatsKey.VM_GC_STATS_COLLECTION_TIME).intValue());
+  }
+
+  void increaseStats(String name, Number value) {
+    if (name.equals(StatsKey.VM_GC_STATS_COLLECTIONS)) {
+      collections.set(collections.get() + value.longValue());
```

Comment:
to make this thread safe do this:
collection.getAndAdd(value.longValue());

Reply:
👍